storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManager.java

```diff
@@ -210,4 +210,14 @@ void onPartitionLeadershipChanges(Set<TopicIdPartition> leaderPartitions,
      * @return Total size of the log stored in remote storage in bytes.
      */
     long remoteLogSize(TopicIdPartition topicIdPartition, int leaderEpoch) throws RemoteStorageException;
-}
\ No newline at end of file
+
+    /**
+     * Denotes whether the partition metadata is ready to serve.
+     *
+     * @param topicIdPartition topic partition
+     * @return True if the partition is initialized for remote storage operations.
+     */
+    default boolean isInitialized(TopicIdPartition topicIdPartition) throws RemoteStorageException {
```

Comment:
This is a backward compatible change and RLMM plugin users do not need to implement it for now as it has the default implementation. We can raise a KIP post 3.6.